File: csharp/Game/Character.cs
namespace Game {
    public abstract class Character(string name, int health)
    {
        // In C#, properties are used instead of getter methods like getName() in Java.
        // The 'Name' and 'Health' properties are read-only and initialized via the constructor.
        // This is different from Java, where fields are typically private and accessed via methods.
        public string Name { get; } = name;
        public int Health { get; } = health;
        

        public abstract void Attack();

        public bool IsAlive() => Health > 0;

    }
}

File: java/game/src/main/java/example/Character.java
package example;

// This abstract class showcases inheritance and polymorphism in Java.
// It defines common properties and methods for all characters, such as name, health, and the isAlive method.
public abstract class Character {
    protected String name;
    protected int health;

    protected Character(String name, int health) {
        this.name = name;
        this.health = health;
    }

    public abstract void attack();

    public boolean isAlive() {
        return this.health > 0;
    }

    public String getName() {
        return name;
    }

    public int getHealth() {
        return health;
    }
}
